tests/unit/Core/Checkout/Order/SalesChannel/OrderRouteTest.php

```diff
@@ -93,4 +97,103 @@ public function testLoadCustomerOrder(): void
 
         static::assertSame($order->getId(), $responseOrder->getId());
     }
+
+    /**
+     * @param ?class-string<\Throwable> $exception
+     */
+    #[DataProvider('customerDataProvider')]
+    public function testValidateGuestCustomer(?bool $isGuest, ?string $mail, ?string $postalCode, ?string $exception): void
+    {
+        if ($exception !== null) {
+            $this->expectException($exception);
+        }
+
+        $orderCustomer = new OrderCustomerEntity();
+        $orderCustomer->setId(Uuid::randomHex());
+        $orderCustomer->setEmail('test@example.com');
+
+        if ($isGuest !== null) {
+            $customer = new CustomerEntity();
+            $customer->setId(Uuid::randomHex());
+            $customer->setGuest($isGuest);
+
+            $orderCustomer->setCustomer($customer);
+        }
+
+        $billingAddress = new OrderAddressEntity();
+        $billingAddress->setZipcode('AA-345');
+
+        $order = new OrderEntity();
+        $order->setId(Uuid::randomHex());
+        $order->setCreatedAt(new \DateTime());
+        $order->setOrderCustomer($orderCustomer);
+        $order->setBillingAddress($billingAddress);
+
+        $context = $this->createMock(SalesChannelContext::class);
+        $context
+            ->method('getCustomer')
+            ->willReturn(null);
+
+        $eventDispatcher = $this->createMock(EventDispatcherInterface::class);
+        $eventDispatcher
+            ->expects($this->once())
+            ->method('dispatch')
+            ->willReturnCallback(function (object $event): object {
+                static::assertInstanceOf(OrderCriteriaEvent::class, $event);
+
+                return $event;
+            });
+
+        $searchResult = new EntitySearchResult(
+            OrderDefinition::ENTITY_NAME,
+            1,
+            new OrderCollection([$order]),
+            null,
+            new Criteria(),
+            Context::createDefaultContext()
+        );
+
+        $orderRepository = $this->createMock(EntityRepository::class);
+        $orderRepository
+            ->expects($this->once())
+            ->method('search')
+            ->willReturn($searchResult);
+
+        $route = new OrderRoute(
+            $orderRepository,
+            $this->createMock(EntityRepository::class),
+            $this->createMock(RateLimiter::class),
+            $eventDispatcher,
+        );
+
+        $criteria = new Criteria();
+        $criteria->addFilter(new EqualsFilter('deepLinkCode', 'deepLinkCode'));
+
+        $request = new Request();
+        $request->attributes->set('email', $mail);
+        $request->attributes->set('zipcode', $postalCode);
+
+        /** @var OrderEntity $responseOrder */
```

Comment:
please do not use var annotations. Especially in tests you can simply assert the type. also see https://github.com/shopware/shopware/blob/trunk/coding-guidelines/core/writing-code-for-static-analysis.md